website/static/js/share/results.js

```diff
@@ -1,126 +1,252 @@
+        ]);
+
+    }
+};
+
+var Subject = {
+    view: function(ctrl, params) {
+        var subject = params.subject;
+        var vm = params.vm;
+        return m('span', m('.badge.pointer', {onclick: function(){
+                utils.updateFilter(vm, 'match:subjects:"' + subject + '"', true);
+            }}, subject.length < 50 ? subject : subject.substring(0, 47) + '...'),
+            ' ');
+    }
+};
+
+var Footer = {
+    controller: function(params) {
+        var self = this;
+        this.result = params.result;
+        this.cleanResult = m.prop(null);
+        this.loadRawNorm = function() {
+            utils.loadRawNormalized(this.result).then(
+                function(cleanData) {
+                    self.cleanResult(cleanData);
+                }, function(error) {
+                    self.cleanResult(error);
+                }
+            );
+        };
+    },
+    view: function(ctrl, params) {
+        var result = params.result;
+        var vm = params.vm;
+        return m('div', [
+            m('span',
+                'Released on ' + new $osf.FormattableDate(result.providerUpdatedDateTime).local,
+                vm.rawNormedLoaded() ?  m('span', [
+                    m('span', {style: {'margin-right': '5px', 'margin-left': '5px'}}, ' | '),
+                    m('a', {
+                        onclick: function() {
+                            ctrl.showRawNormed = ctrl.showRawNormed ? false : true;
+                            ctrl.loadRawNorm();
+                        }
+                    }, 'Data')
+                ]) : m('span', [])
+            ),
+            m('span.pull-right', [
+                m('img', {src: vm.ProviderMap[result.shareProperties.source].favicon, alt: 'favicon for ' + result.shareProperties.source, style: {width: '16px', height: '16px'}}),
+                ' ',
+                m('a', {onclick: function() {utils.updateFilter(vm, 'shareProperties.source:' + result.shareProperties.source);}}, vm.ProviderMap[result.shareProperties.source].long_name),
+                m('br')
+            ]),
+            ctrl.showRawNormed ? m.component(RawNormalizedData, {result: ctrl.cleanResult}) : '',
+        ]);
+    }
+};
+
+var RawNormalizedData = {
+    view: function(ctrl, params) {
+        var result = params.result();
+        var divID = (result.normalized.shareProperties.docID + result.normalized.shareProperties.source).replace( /(:|\.|\[|\]|,)/g, '-' );
+        return m('.row', [
+            m('.col-md-12',
+                m('div', [
+                    m('ul', {class: 'nav nav-tabs'}, [
+                        m('li', m('a', {href: '#raw' + divID, 'data-toggle': 'tab'}, 'Raw')),
+                        m('li', m('a', {href: '#normalized' + divID, 'data-toggle': 'tab'}, 'Normalized'))
+                    ]),
+                    m('div', {class: 'tab-content'},
+                        m('div',
+                            {class: 'tab-pane active', id:'raw' + divID},
```

Comment:
They're necessary to define the correct tab to display when the raw normalized results are shown - before they were simply "raw" and "normalized" for every result - but that caused a weird issue when 2 data tabs were open at once on the page, as they all had the same CSS ids.  Clicking on the "raw" or "normalized" tabs of a further down result would bounce you back up to the first result. With unique CSS IDs per result, you can have multiple data displays open, and still tab between the raw and normalized results of the data tab you're currently viewing.